application/backup.sh

```diff
@@ -7,6 +7,31 @@ log() {
     echo "[$(date '+%Y-%m-%d %H:%M:%S')] ${level}: ${message}"
 }
 
+# Function to detect the appropriate database dump command
+function get_database_dump_command() {
+    MARIADB=$(which mariadb-dump 2>/dev/null)
+    MYSQL=$(which mysqldump 2>/dev/null)
+    MARIADB_STAT=$(stat $MARIADB 2>/dev/null | grep file)
+    MYSQL_STAT=$(stat $MYSQL 2>/dev/null | grep file)
+
+    # If both exist, prefer mysqldump for MySQL RDS
+    if [ -n "$MYSQL_STAT" ] && [ -n "$MARIADB_STAT" ]; then
+        DATABASE_DUMP_COMMAND="mysqldump";
+        log "INFO" "Both mariadb-dump and mysqldump exist, using ${DATABASE_DUMP_COMMAND} for MySQL RDS"
+    elif [ -z "$MYSQL_STAT" ] && [ -n "$MARIADB_STAT" ]; then
+        DATABASE_DUMP_COMMAND="mariadb-dump";
+        log "INFO" "mariadb-dump exists, using ${DATABASE_DUMP_COMMAND} for MariaDB"
+    elif [ -n "$MYSQL_STAT" ] && [ -z "$MARIADB_STAT" ]; then
+        DATABASE_DUMP_COMMAND="mysqldump";
+        log "INFO" "mysqldump exists, using ${DATABASE_DUMP_COMMAND} for MySQL"
+    else
```

Comment:
One of the condition checks can be eliminated if you start with `[ -n "$MYSQL_STAT" ]`. If MySQL is available, just use mysqldump without checking for MariaDB, then if MariaDB is available, use it. If neither, then error.